src/main/python/org/broadinstitute/hellbender/gcnvkernel/io/io_commons.py

```diff
@@ -16,23 +17,71 @@
 _logger = logging.getLogger(__name__)
 
 
+def read_csv(input_file: str,
+             dtypes_dict: Dict[str, object]=None,
+             mandatory_columns_set: Set[str]=None,
+             comment=io_consts.default_comment_char,
+             delimiter=io_consts.default_delimiter_char) -> pd.DataFrame:
+    """Opens a file and seeks to the first line that does not start with the comment character,
+       checks for mandatory columns in this column-header line, and returns a pandas DataFrame.
+       Prefer using this method rather than pandas read_csv, because the comment character will only have an effect 
+       when it is present at the beginning of a line at the beginning of the file (pandas can otherwise strip 
+       characters that follow a comment character that appears in the middle of a line, which can corrupt sample names).
+       Dtypes for columns can be provided, but those for column names that are not known ahead of time will be inferred.
+
+    Args:
+        input_file: input file
+        dtypes_dict: dictionary of column headers to dtypes; keys will be taken as mandatory columns unless
+                     mandatory_columns_set is also provided
+        mandatory_columns_set: set of mandatory header columns; must be subset of dtypes_dict keys if provided
+        comment: comment character
+        delimiter: delimiter character
+
+    Returns:
+        pandas DataFrame
+    """
+    with open(input_file, 'r') as fh:
+        while True:
+            pos = fh.tell()
+            line = fh.readline()
+            if not line.startswith(comment):
+                fh.seek(pos)
+                break
+        input_pd = pd.read_csv(fh, delimiter=delimiter, dtype=dtypes_dict)  # dtypes_dict keys may not be present
```

Comment:
Not sure I'm completely following this. It looks like this reads the csv using pandas starting from the beginning of the column header line. I see that you provide the expected datatypes for the columns, but how does this avoid the midline comment character issue? That is, what happens if sample ids containing the comment character are present in the column header line? Or is that never the case?

Reply:
We call `pd.read_csv` without specifying the `comment` parameter, in which case it defaults to `None`, so there's no effect.

In any case, we currently don't encounter the situation you describe in any of our files (but it's not too hard to imagine that we might in a future file format).